modules/rke2/cloud-config.tpl.yaml

```
@@ -0,0 +1,21 @@
+#cloud-config
+hostname: ${hostname}.local
+prefer_fqdn_over_hostname: false
+users:
+  - name: nutanix
+    sudo: ['ALL=(ALL) NOPASSWD:ALL']
+    shell: /bin/bash
+    ssh_authorized_keys:
+    %{ for key in authorized_keys ~}
+    - ${key}
+    %{ endfor ~}
+ssh_pwauth: False
+runcmd:
+  - |
+    # If no bootstrap IP is provided then start RKE2 as single node/bootstrap
+    if [ -z ${bootstrap_ip} ]; then
+      server_ip=$(ip route get $(ip route show 0.0.0.0/0 | grep -oP 'via \K\S+') | grep -oP 'src \K\S+')
+    else
+      server_ip=${bootstrap_ip}
```

```suggestion
      server_ip=${bootstrap_ip}
      # If server node(s) are ready then an unauthenticated request to the kube api should return a 401
      until $(curl --silent -I https://$${server_ip}:6443 -k | head -n 1|cut -d$' ' -f2) == 401; do
        echo "Waiting for cluster to be available."
        sleep 5
      done
```
Before running rke2-startup.sh wait for the join host to actually be available. This will prevent the rke2 service from starting in an error state that causes all of the rke2-startup.sh steps that run after the service starts to fail.